action/protocol/rewarding/protocol.go

```diff
@@ -396,28 +419,35 @@ func (p *Protocol) settleAction(
 			return nil, err
 		}
 	}
-	gasFee := big.NewInt(0).Mul(actionCtx.GasPrice, big.NewInt(0).SetUint64(actionCtx.IntrinsicGas))
-	depositLog, err := DepositGas(ctx, sm, gasFee)
-	if err != nil {
-		return nil, err
-	}
-	if depositLog != nil {
-		tLogs = append(tLogs, depositLog)
-	}
-	if err := p.increaseNonce(sm, actionCtx.Caller, actionCtx.Nonce); err != nil {
-		return nil, err
+	if !isSystemAction || !protocol.MustGetFeatureCtx(ctx).SkipUpdateForSystemAction {
+		gasFee := big.NewInt(0).Mul(actionCtx.GasPrice, big.NewInt(0).SetUint64(actionCtx.IntrinsicGas))
+		depositLog, err := DepositGas(ctx, sm, gasFee)
+		if err != nil {
+			return nil, err
+		}
+		if depositLog != nil {
+			tLogs = append(tLogs, depositLog)
+		}
+		if err := p.increaseNonce(ctx, sm, actionCtx.Caller, actionCtx.Nonce, isSystemAction); err != nil {
+			return nil, err
+		}
 	}
 	return p.createReceipt(status, blkCtx.BlockHeight, actionCtx.ActionHash, actionCtx.IntrinsicGas, logs, tLogs...), nil
 }
 
-func (p *Protocol) increaseNonce(sm protocol.StateManager, addr address.Address, nonce uint64) error {
-	acc, err := accountutil.LoadOrCreateAccount(sm, addr)
+func (p *Protocol) increaseNonce(ctx context.Context, sm protocol.StateManager, addr address.Address, nonce uint64, isSystemAction bool) error {
```

Comment:
we don't need to change this func? before SkipUpdateForSystemAction HF, it work as in current code, after SkipUpdateForSystemAction HF, this func won't be called for system action

Reply:
and I don't quite understand your reply in closed PR3407

Reply:
it has been closed

Reply:
right, that PR is closed, but the same change is moved to this PR, so still have this comment/question